### PR TITLE
Weekly Patch Release v1.5.10

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

```console
gh pr list -s merged -S 'merged:2022-01-19T18:30:00.000Z..2022-02-08T18:30:00.000Z' --json mergedAt,milestone,url,mergeCommit,title --jq 'sort_by(.mergedAt) | reverse | .[] | select((.milestone.title == "1.5.x") or (.milestone.title == null)) | [.mergedAt, .url, .mergeCommit.oid, .title] | join(" ")' --limit 100
```

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @tchaton @rohitgr7 @akihironitta @borda

### CODE REVIEW

**Comment by ananthsub** [comment]
This should include https://github.com/PyTorchLightning/pytorch-lightning/pull/11776

**Comment by rohitgr7** [comment]
next commit that should be cherry-picked is `11e1e3e6f274177a0d8d36f35d591d8cc2aa6cf7`which will remove the current failures but with introduce failing tests. If anyone knows how to overcome this issue, feel free to push changes here. I can't seem to find the reason for failures.

**Comment by carmocca** [comment]
@rohitgr7 the signal failure is fixed by this PR: https://github.com/PyTorchLightning/pytorch-lightning/pull/11635

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,21 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
+## [1.5.10] - 2022-02-08
```
Comment (carmocca) [inline CHANGELOG.md:7]: Missing:

https://github.com/PyTorchLightning/pytorch-lightning/pull/11481 b8d2c65a372e5d01cfbeace72d6ebd024541ba86 Set the state before saving "last" or "none" checkpoints

Comment (carmocca) [inline CHANGELOG.md:7]: Also:

https://github.com/PyTorchLightning/pytorch-lightning/pull/11672 b8e360dafa77fa9da397159391d3f46be149cc9a [CLI] Fix bug that forces overriding `configure_optimizers`

Comment (rohitgr7) [reply CHANGELOG.md:7]: can you cherry-pick this? I was getting failures.

Comment (rohitgr7) [reply CHANGELOG.md:7]: same here.

Comment (carmocca) [reply CHANGELOG.md:7]: Okay. It's was only a problem in master. I've updated the milestone

Comment (carmocca) [reply CHANGELOG.md:7]: Done!

In the future, you should either ping authors or change the merged PR milestone

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by kaushikb11 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**